pkg/pulumiyaml/diags/diags_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diags
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNonExistentFieldFormatter_fieldsName(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		formatter NonExistentFieldFormatter
+		want      string
+	}{
+		{NonExistentFieldFormatter{FieldsAreProperties: true}, "properties"},
+		{NonExistentFieldFormatter{FieldsAreProperties: false}, "fields"},
+	}
+	for _, tt := range tests {
+		assert.Equalf(t, tt.want, tt.formatter.fieldsName(), "fieldsName()")
+	}
+}
+
+func TestNonExistentFieldFormatter_messageHeader(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		formatter  NonExistentFieldFormatter
+		fieldLabel string
+		want       string
+	}{
+		{NonExistentFieldFormatter{ParentLabel: "parent"}, "label", "label does not exist on parent"},
+	}
+	for _, tt := range tests {
+		assert.Equalf(t, tt.want, tt.formatter.messageHeader(tt.fieldLabel), "messageHeader(%s)", tt.fieldLabel)
+	}
+}
+
+func TestNonExistentFieldFormatter_messageBody(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		formatter NonExistentFieldFormatter
+		field     string
+		want      string
+	}{
```

Comment:
Nit(optional): It seems like it's worth generalizing this test style, since we see the same `tests` struct in most of the new tests.

Reply:
I auto-generated most of this from my IDE, so I'll leave it alone for now in the interest of time.